src/SharedFund.sol

```diff
@@ -48,7 +55,17 @@ contract SharedFund is ERC721 {
         require(_isApprovedOrOwner(_msgSender(), tokenId), "ERC721: caller is not token owner or approved");
 
         _transfer(from, to, tokenId);
+
+        //Update list and map
+        delete owners[from];
+        for (uint i = 0; i < ownersList.length; i++) {
```

Comment:
instead of doing this you should just do 
`ownersList[I] = to`

once fixed you can merge

Reply:
Smart

Reply:
Agree with @msaug